services/queue/src/api.js

```diff
@@ -309,6 +309,117 @@ builder.declare({
+  }
+
+  // Construct status object
+  let status = task.status();
+
+  // If the last run was canceled, resolve dependencies and publish message
+  if (run.state === 'exception' && run.reasonResolved === 'canceled') {
+    // Update dependency tracker
+    await ctx.queueService.putResolvedMessage(
+      task.taskId,
+      task.taskGroupId,
+      task.schedulerId,
+      'exception',
+    );
+
+    // Publish message about the exception
+    const runId = task.runs.length - 1;
+    await ctx.publisher.taskException(_.defaults({
+      status,
+      runId,
+    }, _.pick(run, 'workerGroup', 'workerId')), task.routes);
+    ctx.monitor.log.taskException({ taskId: task.taskId, runId });
+  }
+
+  return status;
+};
+
+/** Cancel all tasks in a group */
+builder.declare({
+  method: 'post',
+  route: '/task-group/:taskGroupId/cancel',
+  name: 'cancelTaskGroup',
+  scopes: 'queue:cancel-task-group:<schedulerId>/<taskGroupId>',
+  stability: APIBuilder.stability.experimental,
+  category: 'Tasks',
+  input: undefined,
+  output: 'cancel-task-group-response.yml',
+  title: 'Cancel Task Group',
+  description: [
+    'This method will cancel all unresolved tasks (`unscheduled`, `pending` or `running` states)',
+    'with the given `taskGroupId`. Behaviour is similar to the `cancelTask` method.',
+    '',
+    'It is only possible to cancel a task group if it has been sealed using `sealTaskGroup`.',
+    'If the task group is not sealed, this method will return a 409 response.',
+    '',
+    'Every task that was canceled with will trigger a `task-exception` message.',
+    'It is possible to rerun cancelled task which will result in a new run.',
+    'Calling `cancelTaskGroup` again in this case will only cancel the new run.',
+    'Other tasks that were already cancelled would not be canceled again.',
+  ].join('\n'),
+}, async function(req, res) {
+  const taskGroupId = req.params.taskGroupId;
+
+  const taskGroup = await TaskGroup.get(this.db, taskGroupId);
+  if (!taskGroup) {
+    return res.reportError('ResourceNotFound',
+      'No task-group with taskGroupId: `{{taskGroupId}}`', {
+        taskGroupId,
+      },
+    );
+  }
+  if (!taskGroup.sealed) {
+    return res.reportError('RequestConflict',
+      'Cannot cancel unsealed task group: `{{taskGroupId}}`', {
+        taskGroupId,
+      },
+    );
+  }
+
+  await req.authorize({
+    taskGroupId,
+    schedulerId: taskGroup.schedulerId,
+  });
+
+  // this will iterate and cancel all tasks that can be canceled and return all tasks
+  const [allTasks, taskGroupSize] = await Promise.all([
+    this.db.fns.cancel_task_group(taskGroupId, 'canceled'),
+    this.db.fns.get_task_group_size(taskGroupId),
+  ]);
+
+  const response = {
+    taskGroupSize: taskGroupSize[0].get_task_group_size,
+    taskIds: [],
+    cancelledCount: 0,
+    taskGroupId,
+  };
+
+  for (let task of allTasks) {
+    task = Task.fromDb(task);
+    await cancelSingleTask(task, this);
```

Comment:
Canceling is done in the DB directly, this is only to notify pulse subscribers and tell queue service that this task was resolved with exception, so it can update tracked dependencies. 

I don't think it should be a big hit on performance, but I didn't test it on tasks groups with thousands tasks.
The call itself might take a bit longer, but for sure faster than all individual task cancel api calls. I can try to put some stress test to see